apis/appmesh/v1beta2/virtualgateway_types.go

```diff
@@ -17,25 +17,262 @@ limitations under the License.
 package v1beta2
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+type VirtualGatewayConditionType string
+
+const (
+	// VirtualGatewayActive is True when the AppMesh VirtualGateway has been created or found via the API
+	VirtualGatewayActive VirtualGatewayConditionType = "VirtualGatewayActive"
+)
+
+// +kubebuilder:validation:Enum=grpc;http;http2
+type VirtualGatewayPortProtocol string
+
+// VirtualGatewayPortMapping refers to https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_PortMapping.html
+type VirtualGatewayPortMapping struct {
```

Comment:
did we introduced different structure for these(e.g. healthCheckPolicy/PortMapping) in the SDK JSON?

The types here should be closely mapping to the types in SDK SON. (except use same case for Acronym, e.g. Tls->TLS). So that we can write tools to generate it in the future.

Reply:
Yes. We have introduced separate structures for Virtual Gateway and Virtual Node in the AWS SDK. Refer [here](https://raw.githubusercontent.com/aws/aws-app-mesh-roadmap/master/appmesh-preview/service-model.json) for details.